pkgs/development/python-modules/glances/default.nix

```diff
@@ -20,19 +20,38 @@ buildPythonPackage rec {
   };
 
   # Some tests fail in the sandbox (they e.g. require access to /sys/class/power_supply):
-  patches = lib.optional doCheck ./skip-failing-tests.patch
-    ++ [ (fetchpatch {
-      # Correct unitest
-      url = "https://github.com/nicolargo/glances/commit/abf64ffde31113f5f46ef286703ff061fc57395f.patch";
-      sha256 = "00krahqq89jvbgrqx2359cndmvq5maffhpj163z10s1n7q80kxp1";
-    }) ];
+  patches = lib.optional (doCheck && stdenv.isLinux) ./skip-failing-tests.patch
+    ++ [
+      (fetchpatch {
+        # Correct unitest
+        url = "https://github.com/nicolargo/glances/commit/abf64ffde31113f5f46ef286703ff061fc57395f.patch";
+        sha256 = "00krahqq89jvbgrqx2359cndmvq5maffhpj163z10s1n7q80kxp1";
+      })
+
+      (fetchpatch {
+        # Fix IP plugin initialization issue
+        url = "https://github.com/nicolargo/glances/commit/48cb5ef8053d823302e7e53490fb22cec2fabb0f.patch";
+        sha256 = "1590qgcr8w3d9ddpgd9mk5j6q6aq29341vr8bi202yjwwiv2bia9";
+      })
+    ];
+
+  # On Darwin this package segfaults due to mismatch of pure and impure
+  # CoreFoundation. This issues was solved for binaries but for interpreted
+  # scripts a workaround below is still required.
+  # Relevant: https://github.com/NixOS/nixpkgs/issues/24693
+  makeWrapperArgs = lib.optional stdenv.isDarwin [
+    "--set" "DYLD_FRAMEWORK_PATH" "/System/Library/Frameworks"
+  ];
 
   doCheck = true;
   checkInputs = [ unittest2 ];
```

Comment:
```suggestion
  checkInputs = [ unittest2 psutil ];
```

Reply:
I'm sorry, I didn't realize `nixpkgs-review` runs build in a sandbox. 2.7 build succeeded there only because it was already built without sandbox during my testing.

On Darwin with sandbox all builds (2.7, 3.7, 3.8) fail in the same way, even if I add `psutil` to `checkInputs`. Do you happen to know if sandbox on Darwin is reliable and endorsed? Should I dig deeper to make these builds work there?

Reply:
[OfBorg built packages on Darwin](https://logs.nix.ci/?attempt_id=147d1f8c-1a83-4e19-9751-925ea7d778b2&key=nixos%2Fnixpkgs.78404) just fine, so I'm guessing these failures have something to do with my env.